Upgrade dependencies only to compatible versions

julienw opened the pull request:
This is the result of a `yarn upgrade`.

I looked closely at the security alerts and I think this should nearly remove all of them.
The last bit would be upgrading flow-coverage-report once https://github.com/rpl/flow-coverage-report/pull/192 is merged, we can do that in a second step.

gregtatum commented:
Ok, here is the error, which looks consistent.

```
Error: [BABEL] /home/circleci/project/src/index.js: Could not find plugin "proposal-numeric-separator". Ensure there is an entry in ./available-plugins.js for it. (While processing: "/home/circleci/project/node_modules/@babel/preset-env/lib/index.js")
    at getPlugin (/home/circleci/project/node_modules/@babel/preset-env/lib/index.js:67:11)
    at Array.from.map.pluginName (/home/circleci/project/node_modules/@babel/preset-env/lib/index.js:258:62)
    at Array.map (<anonymous>)
    at _default (/home/circleci/project/node_modules/@babel/preset-env/lib/index.js:258:43)
    at /home/circleci/project/node_modules/@babel/helper-plugin-utils/lib/index.js:19:12
    at /home/circleci/project/node_modules/@babel/core/lib/config/full.js:199:14
    at Generator.next (<anonymous>)
    at Function.<anonymous> (/home/circleci/project/node_modules/@babel/core/lib/gensync-utils/async.js:26:3)
    at Generator.next (<anonymous>)
    at step (/home/circleci/project/node_modules/gensync/index.js:254:32)
    at evaluateAsync (/home/circleci/project/node_modules/gensync/index.js:284:5)
    at Function.errback (/home/circleci/project/node_modules/gensync/index.js:108:7)
    at errback (/home/circleci/project/node_modules/@babel/core/lib/gensync-utils/async.js:70:18)
    at async (/home/circleci/project/node_modules/gensync/index.js:183:31)
    at onFirstPause (/home/circleci/project/node_modules/gensync/index.js:209:13)
    at Generator.next (<anonymous>)
    at cachedFunction (/home/circleci/project/node_modules/@babel/core/lib/config/caching.js:68:46)
    at cachedFunction.next (<anonymous>)
    at loadPresetDescriptor (/home/circleci/project/node_modules/@babel/core/lib/config/full.js:292:44)
    at loadPresetDescriptor.next (<anonymous>)
    at recurseDescriptors (/home/circleci/project/node_modules/@babel/core/lib/config/full.js:107:30)
    at recurseDescriptors.next (<anonymous>)
    at loadFullConfig (/home/circleci/project/node_modules/@babel/core/lib/config/full.js:142:6)
    at loadFullConfig.next (<anonymous>)
    at Function.transform (/home/circleci/project/node_modules/@babel/core/lib/transform.js:25:45)
    at transform.next (<anonymous>)
    at step (/home/circleci/project/node_modules/gensync/index.js:262:25)
    at gen.next (/home/circleci/project/node_modules/gensync/index.js:266:13)
    at async.call.value (/home/circleci/project/node_modules/gensync/index.js:216:11)
    at errback.call (/home/circleci/project/node_modules/gensync/index.js:184:28)
    at runGenerator.errback (/home/circleci/project/node_modules/@babel/core/lib/gensync-utils/async.js:72:7)
    at val (/home/circleci/project/node_modules/gensync/index.js:108:33)
    at step (/home/circleci/project/node_modules/gensync/index.js:280:14)
    at gen.next (/home/circleci/project/node_modules/gensync/index.js:266:13)
    at async.call.value (/home/circleci/project/node_modules/gensync/index.js:216:11)
```

julienw commented:
Interesting, this didn't fail locally... I'll look.

julienw commented:
I fixed it by changing how the babel package versions were defined. Previously they were defined with an exact version, and I changed them using a "compatible semver" version definition.

Note: I also did it for the other packages that were defined the same way, but they didn't update with this process, so I didn't especially test them.

julienw commented:
Looks like the upgrade of flow-coverage-report wasn't needed after all!
